subsys/net/lib/sockets/Kconfig

```diff
@@ -146,11 +146,6 @@ config NET_SOCKETS_SOCKOPT_TLS
 	bool "TCP TLS socket option support"
 	imply TLS_CREDENTIALS
 	select MBEDTLS if NET_NATIVE
-	imply MBEDTLS_TLS_VERSION_1_2 if !NET_L2_OPENTHREAD
-	imply MBEDTLS_KEY_EXCHANGE_RSA_ENABLED if !NET_L2_OPENTHREAD
-	imply MBEDTLS_CIPHER_AES_ENABLED if !NET_L2_OPENTHREAD
-	imply PSA_WANT_KEY_TYPE_AES if !NET_L2_OPENTHREAD && PSA_CRYPTO_CLIENT
-	imply PSA_WANT_ALG_CBC_NO_PADDING if !NET_L2_OPENTHREAD && PSA_CRYPTO_CLIENT
```

Comment:
You're likely setting yourself up for a fun time (Kconfig-wise) by removing these defaults.

Reply:
Yes, I reverted this back. These configs are not set to "n" in the `prj.conf` file of my test with a better comment than before. I will re-enable what needed on overlay files